src/analyser_p.h

```diff
@@ -22,6 +24,12 @@ limitations under the License.
 #include "logger_p.h"
 #include "utilities.h"
 
+namespace SymEngine {
+template<class T> class RCP;
+class Basic;
+class Symbol;
+} // namespace SymEngine
+
```

Comment:
Yes, this time it's right to have those declarations in this header file (and `#include <map>`).